packages/table/src/quadrants/tableQuadrantStack.tsx

```diff
@@ -776,13 +789,40 @@ export class TableQuadrantStack extends AbstractComponent<ITableQuadrantStackPro
         }
     };
 
-    private maybeSetQuadrantScrollOffset = (quadrantType: QuadrantType, scrollKey: "scrollLeft" | "scrollTop") => {
+    private maybeSetQuadrantScrollOffset = (
+        quadrantType: QuadrantType,
+        scrollKey: "scrollLeft" | "scrollTop",
+        newOffset?: number,
+    ) => {
         const { scrollContainer } = this.quadrantRefs[quadrantType];
+        const scrollOffset = newOffset != null ? newOffset : this.cache.getScrollOffset(scrollKey);
         if (scrollContainer != null) {
-            scrollContainer[scrollKey] = this.cache.getScrollOffset(scrollKey);
+            scrollContainer[scrollKey] = scrollOffset;
         }
     };
 
+    private maybeSetGlobalScrollOffset = (scrollKey: "scrollLeft" | "scrollTop", offset: number) => {
+        this.quadrantRefs[QuadrantType.MAIN].scrollContainer[scrollKey] = offset;
+        this.cache.setScrollOffset(scrollKey, offset);
+
+        const dependentQuadrantType = scrollKey === "scrollLeft" ? QuadrantType.TOP : QuadrantType.LEFT;
+        this.maybeSetQuadrantScrollOffset(dependentQuadrantType, scrollKey);
+    };
+
+    private updateScrollContainerClientSize(isHorizontal: boolean) {
+        // this function is called 'update' instead of 'set', because a 'set'
+        // function typically takes the new value as a parameter. we avoid that
+        // to keep the isHorizontal logic tree contained within this function.
```

Comment:
nit: comments about the function itself should be above defn. since this is a private method, it's going to be added to docs anyway.